src/app/courses/pages/new-course-page/new-course-page.component.ts

```diff
@@ -1,28 +1,54 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { FormControl, FormGroup, Validators } from '@angular/forms';
-import { Router } from '@angular/router';
+import { ActivatedRoute, Params, Router } from '@angular/router';
+import { Subscription } from 'rxjs';
+import { Course } from '../../models/courses.model';
+import { CoursesService } from '../../services/courses.service';
+
 
 @Component({
   selector: 'app-new-course-page',
   templateUrl: './new-course-page.component.html',
   styleUrls: ['./new-course-page.component.scss'],
 })
-export class NewCoursPageComponent implements OnInit {
+export class NewCoursPageComponent implements OnInit, OnDestroy {
 
   formNewCourse!: FormGroup;
+  course: Course;
+  private sub: Subscription
+
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+    private coursesService: CoursesService,
+     ) {}
 
-  constructor(private router: Router) { }
+  title: string = 'New course';
 
   ngOnInit(): void {
     this.buildForm();
+    this.sub = this.route.params.subscribe(
+      (params: Params) => {
+        if (params['id']){
+          this.title = 'Edit course'
+          this.course = this.coursesService.getCourse(+params['id']);
+        }
+      }
+    );
   }
 
+
   onSubmit(): void {
 
   }
 
   createCourse(): void {
```

Comment:
This method creates and updates a course, so the name of the method should be abstract. You can use 'onFormSubmit' or something similar.